lib/workos/portal/portal.ex

```diff
@@ -8,56 +8,20 @@ defmodule WorkOS.Portal do
   @see https://workos.com/docs/admin-portal/guide
   """
 
-  @doc """
-  Create an organization
-
-  ### Parameters
-  - params (map)
-    - domains (list of strings) List of domains that belong to the organization
-    - name (string) A unique, descriptive name for the organization
-
-  ### Example
-  WorkOS.Portal.create_organization(%{
-    domains: ["workos.com"],
-    name: "WorkOS"
-  })
-  """
-  def create_organization(params, opts \\ [])
-
-  def create_organization(params, opts)
-      when is_map_key(params, :domains) and is_map_key(params, :name) do
-    query = process_params(params, [:domains, :name])
-    post("/organizations", query, opts)
-  end
-
-  def create_organization(_params, _opts),
-    do: raise(ArgumentError, message: "need both domains and name in params")
-
-  @doc """
-  Delete an organization
-
-  ### Parameters
-   - organization_id (string) the id of the organization to delete
-
-  ### Example
-  WorkOS.Portal.delete_organization("organization_12345")
-  """
-  def delete_organization(organization, opts \\ []) do
-    delete("/organizations/#{organization}", %{}, opts)
-  end
-
   @doc """
   Generate a link to grant access to an organization's Admin Portal
 
   ### Parameters
   - params (map)
     - intent (string) The access scope for the generated Admin Portal
-     link. Valid values are: ["sso"]
+    link. Valid values are: ["sso"]
```

Comment:
We support the following values currently:

```
["sso", "dsync", "audit_logs", "log_streams"]
```

Is there a preferred way of checking against an enum value in Elixir? For reference, we validate that the intent matches a supported one in Ruby: https://github.com/workos/workos-ruby/blob/main/lib/workos/portal.rb#L58

Reply:
We could use the Enum module to verify if the given intent is supported, eg: `Enum.member?(["sso", "dsync", "audit_logs", "log_streams"], "sso")`

Reply:
Awesome!

Reply:
So I ended up defining a module attribute called `@generate_link_intents` and then using it as a function guard https://github.com/workos/workos-elixir/pull/17/commits/25a2af956713cc6e5c859be3865202868bf8f822

<img width="504" alt="CleanShot 2022-12-21 at 11 51 24@2x" src="https://user-images.githubusercontent.com/48022589/208933443-45f923d9-4630-433d-a203-7b379294abb1.png">

````ex
 def generate_link(%{intent: intent} = _params, _opts)
      when intent not in @generate_link_intents,
      do:
        raise(ArgumentError,
          message:
            "invalid intent, must be one of the following: sso, dsync, audit_logs or log_streams"
        )
````

Reply:
Nice!